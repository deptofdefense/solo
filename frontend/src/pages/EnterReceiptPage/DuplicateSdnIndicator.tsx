import React from "react";
import { Alert } from "solo-uswds";

interface DuplicateSdnIndicatorProps {
  isDuplicate: boolean;
}

const DuplicateSdnIndicator: React.FC<DuplicateSdnIndicatorProps> = ({
  isDuplicate
}) => {
  return (
    <div className="margin-bottom-1em overflow-hidden">
      {isDuplicate && (
        <Alert
          status={"error"}
          className="margin-top-1"
          heading={"SDN already exists in table"}
        />
      )}
    </div>
  );
};

export default DuplicateSdnIndicator;
