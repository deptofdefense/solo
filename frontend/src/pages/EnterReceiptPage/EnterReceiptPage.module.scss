.error {
  color: #d54309;
}
