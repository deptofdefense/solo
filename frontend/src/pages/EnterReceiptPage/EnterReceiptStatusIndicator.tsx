import React from "react";
import { Alert } from "solo-uswds";

interface IndicatorProps {
  loading: boolean;
  status?: "error" | "success";
  message?: string;
}

const EnterReceiptStatusIndicator: React.FC<IndicatorProps> = ({
  loading,
  status,
  message
}) => (
  <div className="grid-row flex-justify-center flex-12 tablet:flex-4">
    {!loading && status && (
      <Alert status={status} className="margin-top-4" heading={message ?? ""} />
    )}
  </div>
);

export default EnterReceiptStatusIndicator;
