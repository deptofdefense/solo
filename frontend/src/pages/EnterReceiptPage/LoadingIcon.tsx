import React from "react";
import classNames from "classnames";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faSpinner,
  faExclamationCircle,
  faCheck
} from "@fortawesome/free-solid-svg-icons";
import { LoadingStatus } from "solo-types";
import classes from "./EnterReceiptPage.module.scss";

const LoadingIcon: React.FC<LoadingStatus> = ({ loading, error }) => (
  <span>
    <FontAwesomeIcon
      spin={loading}
      icon={loading ? faSpinner : error ? faExclamationCircle : faCheck}
      className={classNames({
        [classes.error]: error
      })}
    />
  </span>
);

export default LoadingIcon;
