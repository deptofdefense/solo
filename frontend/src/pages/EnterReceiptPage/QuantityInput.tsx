import React from "react";
import { Input } from "solo-uswds";

interface QuantityInputProps {
  enteredQuantity?: number;
  onQuantitiyChange: (value: number) => void;
}

const QuantityInput: React.FC<QuantityInputProps> = ({
  enteredQuantity = 0,
  onQuantitiyChange
}) => {
  const onValueChange: React.ChangeEventHandler<HTMLInputElement> = e => {
    const asNum =
      e.currentTarget.value === "" ? 0 : parseInt(e.currentTarget.value);
    if (!Number.isNaN(asNum)) {
      onQuantitiyChange(asNum);
    }
  };

  return (
    <Input
      value={enteredQuantity}
      onChange={onValueChange}
      placeholder="Quantity"
    />
  );
};

export default QuantityInput;
