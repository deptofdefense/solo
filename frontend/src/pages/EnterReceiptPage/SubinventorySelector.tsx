import React from "react";
import { Subinventory, Locator } from "solo-types";
import { Select } from "solo-uswds";

interface SubinventorySelectorProps {
  subinventorys: Omit<Subinventory, "locators">[];
  enteredSubinventoryCode: string;
  onSelectSubinventory: (code: string) => void;
}

export const SubinventorySelector: React.FC<SubinventorySelectorProps> = ({
  subinventorys,
  enteredSubinventoryCode,
  onSelectSubinventory
}) => (
  <Select
    value={enteredSubinventoryCode}
    onChange={e => {
      onSelectSubinventory(e.currentTarget.value);
    }}
    placeholder="Subinventory"
  >
    {subinventorys.map(subinv => (
      <option key={subinv.code} value={subinv.code}>
        {subinv.code}
      </option>
    ))}
  </Select>
);

interface LocatorSelectrProps {
  locators: Locator[];
  enteredLocatorCode: string;
  onSelectLocator: (code: string) => void;
}

export const LocatorSelector: React.FC<LocatorSelectrProps> = ({
  locators,
  enteredLocatorCode,
  onSelectLocator
}) => (
  <Select
    value={enteredLocatorCode}
    onChange={e => {
      onSelectLocator(e.currentTarget.value);
    }}
    placeholder="Locator"
  >
    {locators.map(loc => (
      <option key={loc.code} value={loc.code}>
        {loc.code}
      </option>
    ))}
  </Select>
);
