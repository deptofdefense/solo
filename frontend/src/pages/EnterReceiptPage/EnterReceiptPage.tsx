import React, { useMemo } from "react";
import { Title, Table, SdnInputForm } from "components";
import { Document } from "solo-types";
import { Button } from "solo-uswds";
import createColumns from "./tableColumns";
import EnterReceiptStatusIndicator from "./EnterReceiptStatusIndicator";
import useEnterReceiptDocuments from "./useEnterReceiptDocuments";

const EnterReceiptPage: React.FC = () => {
  const {
    docs,
    addSdn,
    submitAllLoadingStatus,
    submitAll,
    modifyDocument,
    removeDocument
  } = useEnterReceiptDocuments();
  const columns = useMemo(() => createColumns(modifyDocument, removeDocument), [
    modifyDocument,
    removeDocument
  ]);

  return (
    <div className="tablet:margin-x-8 overflow-x-auto">
      <Title>Enter Receipt</Title>
      <EnterReceiptStatusIndicator {...submitAllLoadingStatus} />
      <Table<Document>
        columns={columns}
        data={docs}
        manualPagination={false}
        manualSortBy={false}
      />
      <div className="grid-row flex-align-start flex-justify">
        <SdnInputForm
          onSubmit={addSdn}
          disabled={submitAllLoadingStatus.loading}
        />
        <Button
          onClick={submitAll}
          className="margin-top-1"
          disabled={submitAllLoadingStatus.loading || docs.length < 1}
        >
          Submit All
        </Button>
      </div>
    </div>
  );
};

export default EnterReceiptPage;
