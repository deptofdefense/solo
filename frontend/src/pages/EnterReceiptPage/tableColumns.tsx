import React from "react";
import { Column } from "react-table";
import { Document } from "solo-types";
import LoadingIcon from "./LoadingIcon";

type CreateColumns = () => Column<Document>[];

const createColumns: CreateColumns = () => [
  {
    Header: "Loading",
    Cell: ({
      row: {
        original: { loadingStatus }
      }
    }) => <LoadingIcon {...loadingStatus} />
  },
  {
    Header: "SDN",
    accessor: "sdn"
  },
  {
    Header: "NIIN",
    accessor: "part.nsn",
    id: "niin"
  },
  {
    Header: "Nomenclature",
    id: "nomen",
    accessor: "part.nomen"
  },
  {
    Header: "Quantity",
    id: "quantity",
    accessor: ({ enteredReceivedQty }) => enteredReceivedQty
  },
  {
    Header: "Commodity",
    id: "commodity",
    accessor: "suppadd.desc"
  },
  {
    Header: "Subinventory",
    id: "subinventory",
    accessor: () => "Subinventory"
  },
  {
    Header: "Locator",
    id: "locator",
    accessor: () => "locator"
  }
];

export default createColumns;
