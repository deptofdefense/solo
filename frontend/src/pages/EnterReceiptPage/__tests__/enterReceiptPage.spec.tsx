import React from "react";
import EnterReceiptPage from "../EnterReceiptPage";
import LoadingIcon from "../LoadingIcon";
import { defaultApiResponse } from "solo-types";
import { render, fireEvent, wait } from "test-utils";

describe("EnterReceiptPage Component", () => {
  const fetchMock = jest.fn();
  const defaultDoc = defaultApiResponse.results[0];

  afterEach(() => {
    fetchMock.mockReset();
  });

  it("matches snapshot", async () => {
    fetchMock.mockResolvedValue(defaultApiResponse);
    const { asFragment, queryByText, getByText } = render(
      <EnterReceiptPage />,
      {
        authContext: {
          apiCall: fetchMock
        }
      }
    );
    const submitBtn = getByText("Search");
    fireEvent.click(submitBtn);
    await wait(() => {
      // wait for some data to render before checking snapshot
      expect(fetchMock).toHaveBeenCalled();
      expect(queryByText(defaultDoc.sdn)).toBeInTheDocument();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls api and adds result to table when adding an sdn", async () => {
    fetchMock.mockResolvedValue(defaultApiResponse);
    const { queryByText, getByPlaceholderText, getByText } = render(
      <EnterReceiptPage />,
      {
        authContext: {
          apiCall: fetchMock
        }
      }
    );
    const inputField = getByPlaceholderText("SDN");
    const submit = getByText("Search");
    fireEvent.change(inputField, {
      target: { value: "somesdn" }
    });
    fireEvent.click(submit);
    await wait(() => {
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(fetchMock.mock.calls[0][0]).toEqual("/documents?sdn=somesdn");
      expect(queryByText(defaultDoc.part.nomen)).toBeInTheDocument();
    });
  });

  it("keeps sdn in table on fetch error", async () => {
    fetchMock.mockRejectedValue(new Error());
    const { queryByText, getByPlaceholderText, getByText } = render(
      <EnterReceiptPage />,
      {
        authContext: {
          apiCall: fetchMock
        }
      }
    );
    const inputField = getByPlaceholderText("SDN");
    const submit = getByText("Search");
    fireEvent.change(inputField, {
      target: { value: "badsdn" }
    });
    fireEvent.click(submit);
    await wait(() => {
      expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(queryByText(/^M30300/)).toBeInTheDocument();
    });
  });

  it("only updates associated sdn", async () => {
    fetchMock.mockResolvedValue({
      ...defaultApiResponse,
      results: [{ ...defaultDoc, sdn: "1234" }]
    });
    const { queryByText, queryAllByText, getByText } = render(
      <EnterReceiptPage />,
      {
        authContext: {
          apiCall: fetchMock
        }
      }
    );
    const submit = getByText("Search");
    fireEvent.click(submit);
    await wait(() => {
      expect(queryByText("1234")).toBeInTheDocument();
    });
    fetchMock.mockResolvedValue({
      ...defaultApiResponse,
      results: [
        {
          ...defaultDoc,
          sdn: "6789",
          part: { ...defaultDoc.part, nsn: "differentnsn" }
        }
      ]
    });
    fireEvent.click(submit);
    await wait(() => {
      expect(queryByText("1234")).toBeInTheDocument();
      expect(queryByText("6789")).toBeInTheDocument();
      expect(queryAllByText("differentnsn")).toHaveLength(1);
    });
  });
});

describe("LoadingIcon component", () => {
  it("renders loading icon when loading", async () => {
    const { asFragment } = render(<LoadingIcon loading error={false} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders error icon on fetch error", async () => {
    const { asFragment } = render(
      <LoadingIcon loading={false} error={true} message="some error" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders check icon on fetch success", async () => {
    const { asFragment } = render(
      <LoadingIcon loading={false} error={false} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
