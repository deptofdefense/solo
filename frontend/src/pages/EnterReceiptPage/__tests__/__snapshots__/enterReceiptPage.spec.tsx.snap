// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnterReceiptPage Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="tablet:margin-x-8 overflow-x-auto"
  >
    <h1
      class="usa-prose"
    >
      Enter Receipt
    </h1>
    <table
      class="usa-table grid-col-12 usa-table--borderless"
      role="table"
    >
      <thead>
        <tr
          role="row"
        >
          <th
            colspan="1"
            role="columnheader"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Loading
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              SDN
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              NIIN
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Nomenclature
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Quantity
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Commodity
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Subinventory
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
          <th
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="display-flex flex-justify flex-align-center flex-no-wrap text-no-wrap"
            >
              Locator
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-chevron-up fa-w-14 margin-left-05"
                data-icon="chevron-up"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="visibility: hidden;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
                  fill="currentColor"
                />
              </svg>
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            class="td"
            role="cell"
          >
            <span>
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-check fa-w-16 "
                data-icon="check"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </td>
          <td
            class="td"
            role="cell"
          >
            M3030012341234
          </td>
          <td
            class="td"
            role="cell"
          >
            5430015061999
          </td>
          <td
            class="td"
            role="cell"
          >
            TANK,LIQUID STORAGE
          </td>
          <td
            class="td"
            role="cell"
          >
            2
          </td>
          <td
            class="td"
            role="cell"
          >
            Motor Transportation
          </td>
          <td
            class="td"
            role="cell"
          >
            Subinventory
          </td>
          <td
            class="td"
            role="cell"
          >
            locator
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="grid-row flex-align-start flex-justify"
    >
      <form
        class="display-flex flex-row flex-justify-start flex-align-center flex-no-wrap flex-auto margin-1"
      >
        <input
          class="usa-input"
          placeholder="SDN"
          value=""
        />
        <button
          class="usa-button margin-top-1 margin-left-1 square"
          type="submit"
        >
          Search
        </button>
      </form>
      <button
        class="usa-button margin-top-2"
      >
        Submit All
      </button>
    </div>
    <div
      class="grid-row flex-justify-center flex-12 tablet:flex-4"
    />
  </div>
</DocumentFragment>
`;

exports[`LoadingIcon component renders check icon on fetch success 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-check fa-w-16 "
      data-icon="check"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`LoadingIcon component renders error icon on fetch error 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-exclamation-circle fa-w-16 error"
      data-icon="exclamation-circle"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`LoadingIcon component renders loading icon when loading 1`] = `
<DocumentFragment>
  <span>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-spinner fa-w-16 fa-spin "
      data-icon="spinner"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
        fill="currentColor"
      />
    </svg>
  </span>
</DocumentFragment>
`;
